app/controllers/UserProfileController.java

```diff
@@ -19,15 +24,21 @@
 @Controller("userProfileController")
 public class UserProfileController extends BaseController {
     
-    private UserProfileService userProfileService;
+    protected UserProfileService userProfileService;
```

Comment:
Instead of making these protected, it would be much cleaner to create a mock userProfileService and call the setter (which you already do), and access the the mock through the variables defined in the test.